### ISSUE TITLE
bug(ad): implicit-AD gradient magnitude is ~5% wrong on a physical split-CTM state (direction fine; flat in the FD step, so not an FD artifact)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#860]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Split out of the #667 fix (PR #844), where correcting a test fixture stopped masking it.

## Symptom

`ctm_energy_split_implicit_2site`'s gradient has the right **direction** and the wrong **magnitude** on a physical simple-update state.

Measured on `_build_su_neel(D=2)` at `chi = chi_I = 4`, 12 sampled coordinates, against central differences:

```
cos(g_AD, g_FD) = 0.998757          <- direction fine
rel              = 5.09e-2          <- magnitude ~5% off
```

## It is not a finite-difference artifact

This is the discriminator, so it is worth stating precisely. Sweeping the FD step over four decades:

| eps | 1e-3 | 1e-4 | 1e-5 | 1e-6 |
|---|---|---|---|---|
| cos | 0.998757 | 0.998757 | 0.998757 | 0.998757 |
| rel | 5.482e-02 | 5.482e-02 | 5.482e-02 | 5.482e-02 |

(8 sampled coordinates here rather than the test's 12, hence 5.48e-2 vs 5.09e-2.)

Central-difference **truncation** error scales as `eps^2` — it would fall by 10^6 across this range. Central-difference **roundoff** grows as `eps -> 0` — it would blow up at the bottom. The disagreement does neither: it is identical to four significant figures at every step size. The FD estimate is converged and the AD gradient is simply wrong by ~5%.

## Why it only appears now

`tests/test_split_ctm_2site_ad.py::su_state` used to return bare Vidal `Gamma` tensors with **every bond weight dropped** — a markedly less entangled state, on which the same comparison sat under `1e-3`. #844 converts that fixture to the physical (symmetric-gauge) simple-update tensor, and the defect surfaces. The fixture was wrong; the gate was calibrated against it.

This is the fourth defect this cluster has unmasked by removing a degenerate fixture (cf. #766, #767 and the kernel-parity fixture, all exposed when #746/#768 removed the rank-1 fixed point).

## Relation to #841

Same class — direction valid, scale is not — about two orders of magnitude smaller (5% here vs ~15x there). #841 is the dense 1-site implicit-AD path at D=3; this is the split 2-site path at D=2. Worth checking whether one cause explains both before treating them separately; #841's investigation ruled out `conv_method`, degeneracy and the adjoint choice, and left gauge contamination as the live suspect.

## Where it is gated

`tests/test_split_ctm_2site_ad.py`:

- `test_2site_implicit_grad_fd_direction` — live, passes (cos > 0.99).
- `test_2site_implicit_grad_fd_magnitude` — `xfail(strict=True)` pointing here.

Both score one shared module-scoped comparison, so splitting them costs no extra CTM convergences. **Please do not close this by loosening the 1e-3 threshold** — that is exactly how the old fixture hid it.

## Reproduce

```python
from tests.test_split_ctm_2site import _build_su_neel, _heisenberg_gate
from tenax.algorithms._split_ctm_energy_ad import ctm_energy_split_implicit_2site
from tenax.algorithms._ctm_tensor_convergence import CHECKERBOARD_NEIGHBORS
```

then compare `jax.value_and_grad(loss)(A)` against central differences at several `eps`, with

```python
loss = lambda a: ctm_energy_split_implicit_2site(
    {(0, 0): a, (1, 0): B}, CHECKERBOARD_NEIGHBORS, gate,
    chi=4, chi_I=4, max_iter=80, conv_tol=1e-13, min_iter=2).real
```

The flatness in `eps` is the whole point — a single `eps` cannot tell the two explanations apart.
